### PR TITLE
be more explicit on history default value

### PR DESCRIPTION
**Pull request by le-jeu**

### CODE REVIEW

File: core/code/entity_decode.js [hunk 1]
```diff
@@ -111,9 +109,23 @@ function detailsPortalData(a) {
   }
 };
 
-function extendedPortalData(a, required) {
+// on getEntities, intel send the history unless the user has never interacted with the portal
+// raw data is either a summary or an extended data with null details
+// in the first case, history is forced to 0 if missing
```
Comment (johnd0e) [inline core/code/entity_decode.js]: It is unclear what do you mean by 'first case'.
Please change the wording.

File: core/code/entity_decode.js [hunk 2]
```diff
@@ -111,9 +109,23 @@ function detailsPortalData(a) {
...
+function extendedPortalData(a) {
+  // default value on entities from getEntities
+  if (a.length == SUMMARY_PORTAL_DATA_LENGTH)
+    return {
+      history: parseHistoryDetail(0),
+    }
+  // default value if no extended portal data
+  if (a.length < EXTENDED_PORTAL_DATA_LENGTH)
+    return {
+      history: null,
```
Comment (johnd0e) [inline core/code/entity_decode.js]:
```suggestion
      history: null, // no data
```

File: core/code/entity_decode.js [hunk 3]
```diff
@@ -111,9 +109,23 @@ function detailsPortalData(a) {
...
 
-function extendedPortalData(a, required) {
+// on getEntities, intel send the history unless the user has never interacted with the portal
+// raw data is either a summary or an extended data with null details
+// in the first case, history is forced to 0 if missing
+function extendedPortalData(a) {
+  // default value on entities from getEntities
+  if (a.length == SUMMARY_PORTAL_DATA_LENGTH)
+    return {
+      history: parseHistoryDetail(0),
```
Comment (johnd0e) [inline core/code/entity_decode.js]:
```suggestion
      history: parseHistoryDetail(0), // never interacted with this portal
```

**Review by johnd0e — CHANGES_REQUESTED** [review]

File: core/code/entity_decode.js (hunk 1 above)
Comment (le-jeu) [reply core/code/entity_decode.js]:
```suggestion
// when it is a summary, history is forced to 0
```

File: core/code/entity_decode.js [hunk 4]
```diff
@@ -111,9 +109,23 @@ function detailsPortalData(a) {
...
 
-function extendedPortalData(a, required) {
+// on getEntities, intel send the history unless the user has never interacted with the portal
+// raw data is either a summary or an extended data with null details
+// when it is a summary, history is forced to 0
+function extendedPortalData(a) {
+  // default value on entities from getEntities
+  if (a.length == SUMMARY_PORTAL_DATA_LENGTH)
+    return {
+      history: parseHistoryDetail(0), // never interacted with this portal
```
Comment (johnd0e) [inline core/code/entity_decode.js:119]: I suspect that this is wrong. Summary does not contain info and we should return `null`.

File: core/code/entity_decode.js [hunk 5]
```diff
@@ -111,9 +109,23 @@ function detailsPortalData(a) {
...
+    }
+  // default value if no extended portal data
+  if (a.length < EXTENDED_PORTAL_DATA_LENGTH)
+    return {
+      history: null, // no data
```
Comment (johnd0e) [inline core/code/entity_decode.js:124]: But Extended data always contain history info, and `null` here means that the portal never was interacted.
So we should parse 0 here.

**Review by johnd0e — CHANGES_REQUESTED** [review]

File: core/code/entity_decode.js (hunk 4 above)
Comment (le-jeu) [reply core/code/entity_decode.js:119]: that's the reverse, when you get portal data there are two cases:
 - from getEntities:
   - as a summary if the portal was never interacted
   - as extended (with null details) if the portal was interacted
 - from getPortalDetails:
   - always as details (no extended data)

File: core/code/entity_decode.js (hunk 5 above)
Comment (le-jeu) [reply core/code/entity_decode.js:124]: If we get neither a summary or extended data, that's the result of a details requests that doesn't answer history even if the portal was interacted with.

**Pull request closed without merging**